src/plugins/data/common/search/aggs/agg_configs.ts

```diff
@@ -107,6 +110,17 @@ export class AggConfigs {
     return this.opts.partialRows ?? false;
   }
 
+  public get samplerConfig() {
+    return { probability: this.opts.probability ?? 1, seed: this.opts.samplerSeed };
+  }
+
+  isSamplingEnabled() {
+    return (
+      isSamplingEnabled(this.opts.probability) &&
+      this.getRequestAggs().filter((agg) => !agg.type.hasNoDsl).length > 0
```

Comment:
Trying to follow... 
Could you please explain this condition? 

`this.getRequestAggs().filter((agg) => !agg.type.hasNoDsl).length > 0`

Reply:
That condition is set to detect a specific case when no sub-agg is available for the sampling, therefore it makes no sense to have it. That's the case of `Count` of Records in Lens, where it should just return the number of documents.